pkg/status/taskrunpod.go

```diff
@@ -55,8 +56,16 @@ func UpdateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, resourceLis
 				ContainerName:  s.Name,
 				ImageID:        s.ImageID,
 			})
+		} else {
```

Comment:
Non-blocking idea for a future change in this area: We should prefix sidecar containers with something like `sidecar-` like we do for step containers, to prevent possible collisions, and so we can definitively identify them here (`IsSidecarStep`). In the future we might have some third type of container we might want to identify separately.

(Not for this PR, but if you felt like doing it that'd be fine. Otherwise I can file an issue to track it for later)

Reply:
I will do it in a separate commit. If its on the larger side I will just split it off into a separate PR.